src/core/Akka.Remote/Transport/TestTransport.cs

```diff
@@ -397,29 +396,65 @@ public Task<TOut> Apply(TIn param)
+    ///     communication between them.
     /// </summary>
+    /// <remarks>
+    ///     NOTE: This is a global shared state between different actor systems. The purpose of this class is to allow
+    ///     dynamically
+    ///     loaded TestTransports to set up a shared AssociationRegistry.Extensions could not be used for this purpose, as the
+    ///     injection
+    ///     of the shared instance must happen during the startup time of the actor system. Association registries are looked
+    ///     up via a string key. Until we find a better way to inject an AssociationRegistry to multiple actor systems it is
+    ///     strongly recommended to use long, randomly generated strings to key the registry to avoid interference between
+    ///     tests.
+    /// </remarks>
     public class AssociationRegistry
     {
+        private static readonly ConcurrentDictionary<string, AssociationRegistry> registries =
+            new ConcurrentDictionary<string, AssociationRegistry>();
+
         private readonly ConcurrentStack<Activity> _activityLog = new ConcurrentStack<Activity>();
-        private readonly ConcurrentDictionary<Address, Tuple<TestTransport, Task<IAssociationEventListener>>> _transportTable = new ConcurrentDictionary<Address, Tuple<TestTransport, Task<IAssociationEventListener>>>();
-        private readonly ConcurrentDictionary<Tuple<Address, Address>, Tuple<IHandleEventListener, IHandleEventListener>> _listenersTable = new ConcurrentDictionary<Tuple<Address, Address>, Tuple<IHandleEventListener, IHandleEventListener>>();
 
-        private static ConcurrentDictionary<string, AssociationRegistry> registries = new ConcurrentDictionary<string, AssociationRegistry>();
+        private readonly
+            ConcurrentDictionary<Tuple<Address, Address>, Tuple<IHandleEventListener, IHandleEventListener>>
+            _listenersTable =
+                new ConcurrentDictionary<Tuple<Address, Address>, Tuple<IHandleEventListener, IHandleEventListener>>();
+
+        private readonly ConcurrentDictionary<Address, Tuple<TestTransport, Task<IAssociationEventListener>>>
+            _transportTable = new ConcurrentDictionary<Address, Tuple<TestTransport, Task<IAssociationEventListener>>>();
 
+        /// <summary>
+        /// Retrieves the specified <see cref="AssociationRegistry"/> associated with the <see cref="key"/>.
+        /// </summary>
+        /// <param name="key">The registry key - see the HOCON example for details.</param>
+        /// <returns>An existing or new <see cref="AssociationRegistry"/> instance.</returns>
+        /// <code>
+        ///     akka{
+        ///         remote{
+        ///             enabled-transports = ["akka.remote.test"]
+        ///             test{
+        ///                 registry-key = "SOME KEY"
+        ///             }
+        ///         }
+        ///     }
+        /// </code>
```

Comment:
Since I got burned on this before, added some XML-DOC comments explaining how the `AssociationRegistry` works.